src/Illuminate/Events/Dispatcher.php

```diff
@@ -268,6 +268,27 @@ public function dispatch($event, $payload = [], $halt = false)
         return $halt ? null : $responses;
     }
 
+    /**
+     * Dispatch multiple events and call their listeners.
+     *
+     * @param  array<int|string, mixed>  $events
+     * @return void
+     */
+    public function dispatchMany($events)
+    {
+        foreach ($events as $event => $payload) {
+            // If the key is an integer, we can assume we are dealing with an array and not a hashmap.
```

Comment:
why do we care if there are string keys?

Reply:
To support something similar to the `dispatch('key', 'payload')` syntax. e.g:

```php
Event::dispatchMany([
    'event' => [1, 2, 3]
]);